src/main/java/com/wolox/training/repositories/BookRepository.java

```diff
@@ -0,0 +1,21 @@
+package com.wolox.training.repositories;
+
+
+import com.wolox.training.models.Book;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface BookRepository extends JpaRepository<Book, Long> {
+
+    public Optional<Book> findByAuthor(String author);
+
+    public List<Book> findByTitle(String bookTitle);
+
+    public Optional<Book> findByIsbn(String isbn);
+
+    public List<Book> findByPublisherAndGenreAndYear(String publisher, String genre, String year);
```

Comment:
What if some parameter is null? I recommend using validation in the query, something like that:
@Query("SELECT b FROM Book b"
        + " WHERE (:publisher is null or b.publisher = :publisher)"
        + " AND (:genre is null or b.genre = :genre)"
        + " AND (:year is null or b.genre = :year)")

Reply:
Thanks! I develop this feature in the next card, it's ok?

Reply:
Yes, sorry I am going fast :)